src/qt/rpcconsole.h

```diff
@@ -210,4 +212,56 @@ class PlainCopyTextEdit : public QTextEdit {
     }
 };
 
+class NetworkName : public QLabel
+{
+    Q_OBJECT
+
+public:
+    explicit NetworkName(QWidget *parent = nullptr) : QLabel(parent) {}
+
+    void setText(const QString& text) {
+        m_fullText = text;
+        recalculateElidedText();
+    }
+
+    const QString& fullText() const { return m_fullText; }
+    QString selectedText() const {
+        return m_fullText;
+    }
+
+protected:
+    void resizeEvent(QResizeEvent *event) override {
+        QLabel::resizeEvent(event);
+        recalculateElidedText();
+    }
+
+    // Provide the full challenge when Copy&Paste
+    QMimeData* createMimeDataFromSelection() const {
```

Comment:
the full challenge should be copy and pasted - please test
cc: @luke-jr @pablomartin4btc